STYLE.md

```diff
@@ -29,92 +34,140 @@ This document describes guidelines that people new to the codebase tend not to i
+  > Reason: Quick scan-ability of control flow is useful for checking
+    correctness w.r.t. certain classes of bugs.
 
-  > Exception: Sometimes it's better to put control statements on the same line.
+  > Exception: Sometimes it's better to put control statements on the same
+    line.
 
-1. Lines in new code should be under 90 characters long.
+4. Lines in new code should be under 90 characters long.
 
   > Reason: @mlucy's editor windows are that big.
 
-  > Exception: If it's really annoying to do so, or if you incidentally make a minor edit to code that happens to already break that rule.
+  > Exception: If it's really annoying to do so, or if you incidentally make a
+    minor edit to code that happens to already break that rule.
 
-1.  Header files should be included in the following order:
+5.  Header files should be included in the following order:
   1. Your parent .hpp file (if you are a .cc file),
   2. C system headers (`<math.h>`, `<unistd.h>`, etc),
   3. Standard C++ headers (`<vector>`, `<algorithm>`, etc),
   4. Boost headers, with `"errors.hpp"` included before them, if there are any.
   5. Local headers with full paths (`"rdb_protocol/protocol.hpp"`, etc).
   It's nice if they're sorted alphabetically but not important.
 
-  > Reason:  It's always good to have one .cc file include each .hpp file first, so that you know it includes all of its dependencies.  The Google Style Guide specifies the C/C++/Local order and we blindly assume there's a reason.  errors.hpp defines `BOOST_ENABLE_ASSERT_HANDLER` to make Boost assertion failures get forwarded to the RethinkDB assertion mechanism, so it needs to go before boost headers.
+  > Reason: It's always good to have one .cc file include each .hpp file first,
+    so that you know it includes all of its dependencies.  The Google Style
+    Guide specifies the C/C++/Local order and we blindly assume there's a
+    reason.  errors.hpp defines `BOOST_ENABLE_ASSERT_HANDLER` to make Boost
+    assertion failures get forwarded to the RethinkDB assertion mechanism, so
+    it needs to go before boost headers.
 
-1. Don't use non-const lvalue references, except, perhaps, in return values.
+6. Don't use non-const lvalue references, except, perhaps, in return values.
 
-  > Reason: Somebody calling `foo(bar)` can currently expect `bar` not to be modified.  Pass a pointer to bar, if you want it to be mutable.
+  > Reason: Somebody calling `foo(bar)` can currently expect `bar` not to be
+    modified.  Pass a pointer to bar, if you want it to be mutable.
 
   > Exception: `std::swap` and STL-like `swap` methods, perhaps, and other peculiar situations.
 
-1. Don't make fields whose type is a reference type (const or non-const), and don't turn references into pointers in ways that might surprise the caller.
+7. Don't make fields whose type is a reference type (const or non-const), and
```

Comment:
> Don't make reference-type fields (regardless of const-ness), and

sounds much clearer to me. I was expecting another clause after this due to `whose`. Had to re-read it a few times to understand it.